lib/src/report/template.dart

```diff
@@ -183,6 +183,20 @@ Future<ReportSection> followsTemplate(PackageContext context) async {
               '${repository!.verificationFailure}'));
     }
 
+    final executableFiles = await context.executablesInBinDirectory;
+    for (final e in context.pubspec.executables.entries) {
+      final filename = e.value.endsWith('.dart')
+          ? e.value
+          : (e.value.isEmpty ? '${e.key}.dart' : '${e.value}.dart');
+      if (!executableFiles.contains(filename)) {
+        issues.add(Issue(
+          'Missing or invalid executable file: `bin/$filename`.',
+          suggestion: 'Update `executable` field in `pubspec.yaml` or '
+              'fix the missing or invalid file in the `bin/` directory.',
+        ));
+      }
+    }
```

Comment:
I think this is correct, but I have to ask? Are we sure there isn't a diagnostic covering this already?
And if not, should we consider adding it to the `analyzer` instead?

Reply:
writing `pana: pana2.dart` in pana's `pubspec.yaml` does not reveal anything during analysis. I agree, it could get inside `analyzer` too, and once it does, we may just remove it from here. But until then, it is a cheap check we can just do.